lightning/src/ln/channelmanager.rs

```diff
@@ -8755,13 +8774,13 @@ where
 
 		self.router
 			.create_blinded_paths(recipient, peers, secp_ctx)
-			.and_then(|paths| paths.into_iter().next().ok_or(()))
```

Comment:
We should still error if the message router returns an empty `Vec`. Otherwise, we'd silently not send any invoice requests.

We'll need something like:

```rust
.and_then(|paths| (!paths.is_empty()).then(|| paths).ok_or(()))
```

Reply:
Makes sense! We don't want to silently fail to send any invoice requests.
Introduced the check for empty vec in [f541586](https://github.com/lightningdevkit/rust-lightning/commit/f541586a27a202424eb74b4fddab826adbb1f7a7)

Thanks for the suggestion!